plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/DialogStagingDirectoryProvider.java

```diff
@@ -0,0 +1,27 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Shell;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.Messages;
+
+/**
+ * Prompts the user to select a directory for staging.
+ */
+// TODO should be moved to separate UI bundle (if kept at all in deploy's final implementation)
```

Comment:
I'm not sure if this will be kept in the future. Now it helps working around the lack of more sophisticated way to configure the staging directory. For now I just wanted an interface that the deploy can depend on obtaining the staging location and a quick implementation via this class. I don't expect this class to be hanging around for long.